MediaBrowser.Model/Cryptography/PasswordHash.cs

```diff
@@ -16,70 +16,78 @@ public class PasswordHash
         public byte[] SaltBytes;
         public string Hash;
         public byte[] HashBytes;
-        public PasswordHash(string StorageString)
+        public PasswordHash(string storageString)
         {
-            string[] a = StorageString.Split('$');
-            Id = a[1];
-            if (a[2].Contains("="))
+            string[] SplitStorageString = storageString.Split('$');
+            Id = SplitStorageString[1];
+            if (SplitStorageString[2].Contains("="))
             {
-                foreach (string paramset in (a[2].Split(',')))
+                foreach (string paramset in (SplitStorageString[2].Split(',')))
                 {
                     if (!String.IsNullOrEmpty(paramset))
                     {
-                        string[] fields = paramset.Split('=');
-                        Parameters.Add(fields[0], fields[1]);
+                        string[] fields = paramset.Split('=');
+                        if(fields.Length == 2)
+                        {
+                            Parameters.Add(fields[0], fields[1]);
+                        }

```

Comment:
I think we should at least log a warning or error here as we obviously have some malformed storage string

Reply:
done